src/System.Console/src/System/ConsoleKeyInfo.cs

```diff
@@ -70,7 +70,11 @@ public bool Equals(ConsoleKeyInfo obj)
 
         public override int GetHashCode()
         {
-            return (int)_keyChar | (int)_mods;
+            // For all typical cases we can fit all bits losslessly into the hash code:
+            // _keyChar could be any 16-bit value (though is most commonly ASCII). Use 16 bits.
+            // _key can theoretically be 32-bit, but only has enum defined values up to 255.  Use 8 bits.
+            // _mods can also theoretically be 32-bit, but only has enum defined values up to 4. Use 8 bits.
+            return _keyChar | ((int)_key << 16) | ((int)_mods << 24);
```

Comment:
That said, given this code in the ctor, it'll be extremely uncommon for there to be any value larger than 8-bits:
```C#
        public ConsoleKeyInfo(char keyChar, ConsoleKey key, bool shift, bool alt, bool control)
        {
            // Limit ConsoleKey values to 0 to 255, but don't check whether the
            // key is a valid value in our ConsoleKey enum.  There are a few 
            // values in that enum that we didn't define, and reserved keys 
            // that might start showing up on keyboards in a few years.
            if (((int)key) < 0 || ((int)key) > 255)
            {
                throw new ArgumentOutOfRangeException(nameof(key), SR.ArgumentOutOfRange_ConsoleKey);
            }
```
You'd essentially need to fill in the values of the struct without going through the ctor.